packages/remix-server-runtime/routes.ts

```diff
@@ -75,15 +75,13 @@ export function createStaticHandlerDataRoutes(
               loadContext,
             })
         : undefined,
-      action: route.module.action
-        ? (args) =>
-            callRouteAction({
-              ...args,
-              routeId: route.id,
-              action: route.module.action,
-              loadContext,
-            })
-        : undefined,
+      action: (args) =>
+        callRouteAction({
+          ...args,
+          routeId: route.id,
+          action: route.module.action,
+          loadContext,
+        }),
```

Comment:
Chatted through this and we want to keep toe consistency of a route not having an action (in the react router route tree) if it didn't export an action.

But we have some issues around the 405's making it to the catch boundary currently due to bad assumptions in `queryRoute`.  We lose the visibility into a "thrown" versus "returned" Response in some cases, most notably the 405 "no action found" scenario.  User-thrown or user-returned responses are already handled by `callRouteAction` so we can ignore those.  But we need to surface out router-generated responses so we can add an `X-Remix-Catch` header to them.

Desired Remix API is as follows:

```js
async function handleDataRequestRR(...) {
  try {
    // Requirements here:
    //  - unable to find a route -> throw a 404 response
    //  - unable to find a loader -> wont happen in remix due to route chunk loads
    //  - unable to find an action -> throw a 405 response
    //  - user throws a response -> captured in callRouteAction 
    //  - user returns a response -> we return that response
    let response = await staticHandler.queryRoute(request, routeId);      

    // Process redirects

    return response;
} catch (error) { 
  // All thrown responses to this point are automatically catch boundary errors,
  // they came from the router not user code
  if (error instanceof Response) {
    error.headers.set('X-Remix-Catch', 'yes');        
    return error;
  }

  // Handle 500 error boundary errors
} 
```
```

Reply:
Matt to look into the changes for createStaticHandler to avoid the above ambiguity.